kernel: bump 6.12 to 6.12.62

graysky2 opened the pull request:
Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.12.62

Manually rebased:
  bcm27xx/patches-6.12/950-0358-mmc-sdhci-of-dwcmshc-define-sdio-timeout-clocks.patch

Removed upstreamed:
  backport-6.12/830-v6.18-spi-bcm63xx-fix-premature-CS-deassertion-on-RX-only-transactions.patch

All other patches automatically rebased.

Build system: x86/64
Build-tested: flogic/glinet_gl-mt6000, ramips/tplink_archer-a6-v3, x86/64-glibc
Run-tested: flogic/glinet_gl-mt6000, ramips/tplink_archer-a6-v3, x86/64-glibc

graysky2 committed:
kernel: bump 6.12 to 6.12.62

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.12.62

Manually rebased:
  bcm27xx/patches-6.12/950-0358-mmc-sdhci-of-dwcmshc-define-sdio-timeout-clocks.patch

Removed upstreamed:
  backport-6.12/830-v6.18-spi-bcm63xx-fix-premature-CS-deassertion-on-RX-only-transactions.patch[1]

All other patches automatically rebased.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.62&id=e31194bf494f6900a5f96f55ed194a00e458f8d1

Build system: x86/64
Build-tested: flogic/glinet_gl-mt6000, ramips/tplink_archer-a6-v3, x86/64-glibc
Run-tested: flogic/glinet_gl-mt6000, ramips/tplink_archer-a6-v3, x86/64-glibc

Signed-off-by: John Audia <therealgraysky@proton.me>
Link: https://github.com/openwrt/openwrt/pull/21126
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

graysky2 commented:
I had some time so I refreshed for 6.12.62-rc1. Since it required some manual intervention, I opened a draft PR documenting the PR which will be refreshed when the release version is out. Most of the time, upstreamed patches are not pulled from rc versions and most of the time, manual rebases do not change. This is why it is in draft status.

I updated the commit message to avoid confusion from those wanting to build or check the work.

EPinci commented:
Build system: x86/64
Build-tested: mediatek/filogic
Run-tested: mediatek/filogic (BPI-R4, GL-X3000)

@graysky2 Resolutions are fine on BPI-R4

hauke committed:
kernel: Fix boot of realtek rtl838x

Revert two patches from upstream Linux:
https://github.com/gregkh/linux/commit/135178e90aa43ad949534e1d6e376c4034942caa
https://github.com/gregkh/linux/commit/63a93d1cd6077d79735f804f5a4957bfb240280c

This fixes a boot hang on realtek rtl838x switches.
This is the last printed message:
Inode-cache hash table entries: 8192 (order: 3, 32768 bytes, linear)

Thread on mips mailing list:
https://lore.kernel.org/linux-mips/b35fe4b3-8f42-49f4-a6bf-9f0e56d4050c@hauke-m.de/T/#u

Link: https://github.com/openwrt/openwrt/pull/21166
Signed-off-by: John Audia <therealgraysky@proton.me>
Link: https://github.com/openwrt/openwrt/pull/21126
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

GoetzGoerisch commented:
@graysky2 please pick the revert from here: https://github.com/GoetzGoerisch/openwrt/commit/6209a608dfefc85337d40824d6ddd9e17c471db9

GoetzGoerisch commented:
kernel 6.12.62: with the reverted upstream patches on top of 25.12 instead of main
run-tested: Zyxel GS1900-8-B1

@graysky2 please cherry-pick and refresh Hauke's commit https://github.com/openwrt/openwrt/commit/b8d232d9dca82078b89f3a18da434cfbae641f2b on top here to get this into 25.12-rc1

GoetzGoerisch reviewed (approved):
rebased on top of `main` with the latest realtek commits from @jonasjelonek 

run-tested: Zyxel GS1900-8-B1, HPE 1920-24G (JG924A)

Thank you!
